pkg/asset/cluster/gcp/permissions.go

```diff
@@ -0,0 +1,50 @@
+package gcp
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	api "google.golang.org/api/cloudresourcemanager/v1"
+)
+
+var (
+	searchPermissions = []string{
+		"compute.firewalls.create",
+	}
+)
+
+// CloudPermissions is a struct that will contain the permission fields mapped to a boolean
+// value where the permission either exists or does not for the Service account in regard
+// to a specific project
+type CloudPermissions struct {
+	CreateFirewallRules bool `json:"compute.firewalls.create"`
+}
+
+// ValidatePermissions will pass a set of permissions to the cloud resource manager api and
+// determine which permissions were found from the original list. When a permission is not listed
+// in the returned data, it is not a permission of the current user in the supplied project.
+func ValidatePermissions(project string) (*CloudPermissions, error) {
```

Comment:
I was putting together a model of what we do for AWS. AWS appears to have a list of permissions that have to be met. Do we want this to be configurable?

Reply:
It would be good and it seems like not too much trouble to keep in mind the use case for general permissions checking when designing this function. The general requirement seems to be the same for both cases, so essentially one function should be able to handle both. The `CloudPermissions` data structure is an interesting one and it's not so obvious to me how this would work if we were comparing a larger set of permissions. 

The permissions that are returned by the `TestIAMPermissions` are in a `[]string`, which could lend itself to converting to a set and using [difference](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/sets#String.Difference) to compare with the original.